src/coreclr/jit/block.cpp

```diff
@@ -96,8 +99,8 @@ void EHSuccessorIterPosition::FindNextRegSuccTry(Compiler* comp, BasicBlock* blo
     // Must now consider the next regular successor, if any.
     while (m_remainingRegSuccs > 0)
     {
+        m_curRegSucc = block->GetSucc(m_numRegSuccs - m_remainingRegSuccs, comp);
```

Comment:
The old logic meant we were visiting EH successors of successors in reverse order. This logic was hard to replicate so I've just "fixed" the old `GetAllSuccs` to visit these in the right order. `GetAllSuccs` exists only for validation, so I expect to remove it within a few weeks anyway.